src/main/java/org/assertj/core/error/ShouldHaveCauseSameAs.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.error;
+
+public class ShouldHaveCauseSameAs extends BasicErrorMessageFactory {
+
+  public static ErrorMessageFactory shouldHaveCauseSameAs(Throwable actualCause, Throwable expectedCause) {
+    return new ShouldHaveCauseSameAs(actualCause, expectedCause);
+  }
+
+  private ShouldHaveCauseSameAs(Throwable actualCause, Throwable expectedCause) {
+    super("%nExpecting:%n  <\"%s\">%nand actual:%n  <\"%s\">%nto refer to the same object.", expectedCause, actualCause);
```

Comment:
It is confusing to reuse actual for actual cause, let's reword to
`%nExpecting actual cause reference to be:%n  <\"%s\">%nbut was:%n  <\"%s\">.`